### PR TITLE
Append to auth.txt after each change, don't rewrite

### PR DESCRIPTION
**Pull request by bendeutsch**
Replaces rewriting auth.txt after each change. The auth.txt becomes a chronological log of changes. 

This works because the (unchanged) `read_auth_file` function reads the auth.txt line for line, with later lines clobbering previous ones. In a classic auth.txt, each player only has one line, but in this log-style version, each player may appear several times, with the most current contents last.

This should mitigate #5334 and possibly #6783.

### CODE REVIEW

**Comment by bendeutsch** [comment]
Shortcomings of the code as of now: the auth.txt file is only condensed by rewriting it at the start of the world, when it is initally read. Long-running worlds will see their auth.txt grow longer and longer; it may be useful to explicitly call `save_auth_file` from time to time. However, the actual data is only held in-memory once, so performance during a game should not be affected.

File: builtin/game/auth.lua [hunk 1]
```diff
@@ -53,7 +53,31 @@ local function save_auth_file()
 	end
 end
 
+local function append_auth_file(name)
+	local stuff = auth_table[name]
+	-- Check table for validness before attempting to save
+	assert(type(name) == "string")
+	assert(name ~= "")
+	assert(type(stuff) == "table")
```
Comment (SmallJoker) [inline builtin/game/auth.lua]: `stuff` is nil when called from `delete_auth(...)`

Comment (bendeutsch) [reply builtin/game/auth.lua]: Good point – I hadn't tested `delete_auth`…

… which reminds me that I had wanted to handle deletes, but completely forgot! D'oh!

**Comment by bendeutsch** [comment]
Marked as WIP because when I had mapped this all out in my brain, I had a plan to deal with deletes – and forgot to implement it!

Brb…

**Comment by bendeutsch** [comment]
…there. In the current code, a line of 
```
<name>:-:-:0
```
is inserted when you delete the auth of player <name>. On the next full auth file read, the corresponding `auth_table` entry is set to `nil`, effectively removing them. The full save after that purges the name from the file entirely.

(I had to make my own mini-mod to test this – there isn't any builtin command to "delete an account", is there?)

**Comment by bendeutsch** [comment]
By the way, I'm aware of #6784 and am not trying to circumvent or sidestep it. This is more of a stop-gap measure until a real database version is implemented.

(I might try my hand at it myself, I just don't know how much work I'd be duplicating…)

**Comment by rubenwardy** [comment]
Obsolete with SQLITE auth

**Pull request closed without merging**